OpenRA.Mods.Cnc/Traits/Disguise.cs

```diff
@@ -211,14 +223,58 @@ void HandleDisguise(Player oldEffectiveOwner, bool oldDisguiseSetting)
 
 		void INotifyAttack.Attacking(Actor self, Target target, Armament a, Barrel barrel)
 		{
-			if (info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Attack))
+			if (info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Attack) || info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Demolish))
 				DisguiseAs(null);
 		}
 
 		void INotifyDamage.Damaged(Actor self, AttackInfo e)
 		{
+			if (e.Damage.Value == 0)
```

Comment:
It felt rather funny when I saw it in a few other places that used it. I'll remove it. :)